File: python/simple_pvr/database_initializer.py
from sqlalchemy import Column
from sqlalchemy.orm import sessionmaker, relationship, backref
from sqlalchemy.types import Integer, String, Text, DateTime, Boolean
#require 'active_support/time_with_zone'

import os

from .master_import import Schedule, Programme, Channel, Category, db

db.create_all()

class DatabaseInitializer():

    def setup(self, database_file_name = None):
        if not database_file_name:
            database_file_name = os.pardir() + "/database.sqlite"


    def clear(self):
        Schedule.query.delete()
        Programme.query.delete()
        Channel.query.delete()
        Category.query.delete()

File: ruby/lib/simple_pvr/model/database_initializer.rb
require 'data_mapper'
require 'dm-migrations'
require 'active_support/time_with_zone'
require File.dirname(__FILE__) + '/channel'
require File.dirname(__FILE__) + '/programme'
require File.dirname(__FILE__) + '/schedule'
require File.dirname(__FILE__) + '/recording'

DataMapper.finalize

module SimplePvr
  module Model
    class DatabaseInitializer
      def self.setup(database_file_name = nil)
        database_file_name ||= Dir.pwd + '/database.sqlite'
        DataMapper.setup(:default, "sqlite://#{database_file_name}")
        DataMapper.auto_upgrade!
      end
    
      def self.clear
        Schedule.destroy
        Programme.destroy
        Channel.destroy
      end
    
      def self.prepare_for_test
        return if @initialized
      
        @database_file_name = Dir.pwd + '/spec/resources/test.sqlite'
        File.delete(@database_file_name) if File.exists?(@database_file_name)
        self.setup(@database_file_name)
        @initialized = true
      end
    end
  end
end